undertow/src/main/java/org/wildfly/extension/undertow/filters/FilterDefinitions.java

```diff
@@ -45,8 +45,8 @@ public class FilterDefinitions extends PersistentResourceDefinition {
 
     public FilterDefinitions() {
         super(PATH_ELEMENT, UndertowExtension.getResolver(PATH_ELEMENT.getValue()),
-                new AbstractAddStepHandler(),
-                ReloadRequiredRemoveStepHandler.INSTANCE
+                new ModelOnlyAddStepHandler(),
+                ModelOnlyRemoveStepHandler.INSTANCE
```

Comment:
This remove handler change seems out of scope and perhaps changes the reported read-operation-description behavior. There were some recent bug-fixing changes in this area and I'm reluctant to change this as part of this work.

Reply:
This change is the essence of this PR. This resource has no runtime - thus the existing handlers create a completely unnecessary runtime step.
If this resource contains any child resources (i.e. actual filter resource with runtime handling), removing this resource will recursively remove its children and put the server into reload required.

Reply:
> This remove handler change seems out of scope and perhaps changes the reported read-operation-description behavior.

Can you elaborate?  I don't see any recent changes to the ReadOperationDescriptionHandler.  Yes, this would change the reported "restart-required" value for the remove operation - but that's the whole point.

Reply:
Ok; I see now both of these resources are direct children of the subsystem root and truly have no runtime behavior, i.e. no parent whose behavior is influenced by these.

I believe the thing I was talking about related to FilterRefDefinition, which is a different case.

Thanks, and sorry for the noise.